### PR TITLE
Improve the errors returned during version publish

### PR DESCRIPTION
**Pull request by confused-Techie**
### Requirements 

* Filling out the template is required.
* All new code requires tests to ensure against regressions.
  - However, if your PR contains zero code changes, feel free to select the checkmark below to indicate so.

* [X] Have you ran tests against this code?
* [ ] This PR contains zero code changes.

### Description of the Change

This PR expands the usage of `common.detailedError()` and the new `content` field of ServerStatusObjects to improve the error messages returned to users during the publishing of a new version.

This aims to help users understand what has gone wrong during a version publish, as well as provide further insight into some obfuscated aspects of the request handling. Ideally to allow smoother troubleshooting of these issues.

### CODE REVIEW

File: src/auth.js [hunk 1]
```diff
@@ -90,20 +91,25 @@ async function verifyAuth(token) {
...
+
     return {
       ok: true,
-      content: authUserObject,
+      content: authUserObject
     };
+
   } catch (err) {
-    logger.generic(3, "auth.verifyAuth() Caught an error", {
+    logger.generic(3, "auth.verifyAuth() Cuaght an Error", {
```
Comment (Daeraxa) [inline src/auth.js:102]: Was just skimming through, was this change intentional from `Caught` to `Cuaght`?

Comment (confused-Techie) [reply src/auth.js:102]: Great catch, it absolutely was not lol

Comment (confused-Techie) [inline src/auth.js:102]:
```suggestion
    logger.generic(3, "auth.verifyAuth() Caught an Error", {
```
